lib/index.spec.js

```diff
@@ -60,51 +60,48 @@ describe('DeploymentBucketPlugin', () => {
 
       expect(plugin.config).toEqual({})
     })
+  })
 
-    it('should default versioning to false if missing property "custom.deploymentBucket.versioning"', () => {
+  describe('with configuration enabled = false', () => {
+    beforeEach(() => {
       serverless.service.provider.deploymentBucketObject = {
-        name: 'some-bucket'
+        name: 'some-bucket',
+        serverSideEncryption: 'AES256'
       }
       serverless.service.custom = {
-        deploymentBucket: {}
+        deploymentBucket: {
+          enabled: false
+        }
       }
       plugin = new DeploymentBucketPlugin(serverless, options)
-
-      expect(plugin.config.versioning).toEqual(false)
-    })
-
-    it('should not set hooks if missing property "custom.deploymentBucket.name"', () => {
-      serverless.service.provider.deploymentBucketObject = {}
-      plugin = new DeploymentBucketPlugin(serverless, options)
-
-      expect(plugin.hooks).not.toHaveProperty('before:package:setupProviderConfiguration')
     })
 
-    it('should not set hooks if empty property "custom.deploymentBucket.name"', () => {
-      serverless.service.provider.deploymentBucketObject = {
-        name: ''
-      }
-      plugin = new DeploymentBucketPlugin(serverless, options)
-
-      expect(plugin.hooks).not.toHaveProperty('before:package:setupProviderConfiguration')
+    it('should not set hooks', () => {
+      expect(plugin.hooks).toEqual({})
     })
   })
 
-  describe('with configuration enabled = false', () => {
+  describe('with configuration enabled = false and not empty configuration stages', () => {
     beforeEach(() => {
-      serverless.service.provider.deploymentBucketObject = {
-        name: 'some-bucket',
-        serverSideEncryption: 'AES256'
-      }
       serverless.service.custom = {
         deploymentBucket: {
-          enabled: false
+          enabled: false,
+          stages: ['local']
         }
       }
+    })
+
+    xit('should set hooks', () => {
```

Comment:
In this two scenario I need help :)

Reply:
Once you resolve the comments I've made, I will take a look at what we can do to test and obtain code coverage.